### PR TITLE
Implement support for {x, y} objects in Point#convert

### LINKED ISSUE #3: Support {x, y} objects in Point#convert

### PR DESCRIPTION
**Pull request by bfrengley**
Fixes #3.

### CODE REVIEW

**Comment by mourner** [comment]
Hey, thanks for the PR! Can you provide some context on why you want to add this? I no longer remember the original reason for #3.

**Comment by bfrengley** [comment]
I think the original reason for #3 was to maintain consistency with `LngLat#convert` converting from `{lng, lat}` objects (mapbox/mapbox-gl-js#2322).

We store mouse coordinates in a Redux state tree as `{ x: number; y: number; }`, which doesn't necessarily maintain the `Point` prototype. As such, it would be nice to be able to use these directly where we would use [`PointLike`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mapbox-gl/index.d.ts#L19) without having to convert to an array.

**Comment by Mechazawa** [comment]
Could this still get merged?

**Comment by bfrengley** [comment]
@mourner I've updated this pull request to solve the merge conflicts, so it should be able to be merged to fix mapbox/mapbox-gl-js#8968.

File: index.js [hunk 1]
```diff
@@ -308,5 +310,12 @@ Point.convert = function (a) {
     if (Array.isArray(a)) {
         return new Point(a[0], a[1]);
     }
+    if (
+        a &&
+        Object.prototype.hasOwnProperty.call(a, 'x') &&
+        Object.prototype.hasOwnProperty.call(a, 'y')
```
Comment (Mechazawa) [inline index.js:316]: Why not just `a && 'x' in a && 'y' in a`

**Pull request closed without merging**